source/slang/core.meta.slang

```diff
@@ -74,6 +74,12 @@ syntax snorm : SNormModifier;
 ///
 syntax __extern_cpp : ExternCppModifier;
 
+__magic_type(IDefaultInitializable)
```

Comment:
Can be done separately, but when we have `struct MyType : IDefaultInitializable { int f; }`
we should be able to automatically synthesize a default initializer to make it conform to the interface.

Reply:
This can be done by extending the logic in `trySynthesizeConstructorRequirementWitness`.